.github/workflows/haskell.yml

```diff
@@ -21,6 +21,22 @@ jobs:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
+    - uses: actions/cache@preview
+      name: Cache ~/.cabal/packages
+      with:
+        path: ~/.cabal/packages
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-packages-${{ hashFiles('**/fused-effects.cabal') }}
```

Comment:
Should we exclude e.g. the hackage index since we regenerate it every time with `cabal update`?

Reply:
Yes! Do you know where that file is, and does it exist before we run this cache action?

Reply:
These are the files that we remove with travis in `semantic` (and which we got from the haskell-ci project):

https://github.com/github/semantic/blob/master/.travis.yml#L10-L18

I _think_ they shouldn’t exist until we’ve done a `cabal update` (unless they were restored there via an earlier cache).

Reply:
I'm going to leave this alone for now, we might cache stuff that gets overridden but I think that's ok. In the future, if we want to do this I think it'll be another step like this:

```
    - name: Clear out hackage index
      run: |
        rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.*
        etc...
```